packages/arb-token-bridge-ui/src/components/TransferPanel/hooks/useIsTransferAllowed.ts

```diff
@@ -0,0 +1,52 @@
+import { useMemo } from 'react'
+import { useAccount, useNetwork } from 'wagmi'
+
+import { useAppState } from '../../../state'
+import { useNetworks } from '../../../hooks/useNetworks'
+import { useDestinationAddressError } from './useDestinationAddressError'
+
+export function useIsTransferAllowed() {
+  const {
+    app: {
+      arbTokenBridgeLoaded,
+      arbTokenBridge: { eth }
+    }
+  } = useAppState()
+  const { address: walletAddress, isConnected } = useAccount()
+  // do not use `useChainId` because it won't detect chains outside of our wagmi config
+  const { chain } = useNetwork()
+  const [networks] = useNetworks()
+  const { destinationAddressError } = useDestinationAddressError()
+
+  return useMemo(() => {
+    const isConnectedToTheWrongChain = chain?.id !== networks.sourceChain.id
+
+    if (!arbTokenBridgeLoaded) {
+      return false
+    }
+    if (!eth) {
+      return false
+    }
```

Comment:
These conditions were in the while await Promise loop.

Since the while loop is deleted, they are added here to prevent transfers.